backend/src/core/services/credential.service.ts

```diff
@@ -26,31 +26,29 @@ const upsertCredential = async (
 ): Promise<void> => {
   // If credential doesn't exist, upload credential to secret manager
   try {
-    logger.info(`Updating credential in AWS secrets manager for name=${name}`)
-    // Secrets stored in localstack will not be tagged as no error is thrown by localstack if the secret
-    // does not exists.
+    logger.info('Storing credential in AWS secrets manager')
     await secretsManager
-      .putSecretValue({
```

Comment:
Since our developer IAM policy explicitly deny all SecretsManager actions unless they're tagged with "development", `putSecretValue` will always result in an IAM explicit deny error instead of an `ResourceNotFoundException`. This is because the `putSecretValue` call [does not allow us to pass in tags](https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_PutSecretValue.html#API_PutSecretValue_RequestParameters). 

To resolve this, there's two possible solution I thought of:
1. Allow `DescribeSecret` regardless of environment tag in our developer IAM policy. We will then call this function before `putSecretValue`.  If the secret does not exists, `ResourceNotFoundException` will be thrown. 
2. Switch around `putSecretValue` and `createSecret`. Since `createSecret` allows us to pass in tags, the call will not be blocked by our IAM policy. 

Decided to go with option 2. While `DescribeSecret` does not return encrypted values, I think it would still be good to restrict access.